test/unit/org/apache/cassandra/cql3/ast/Conditional.java

```diff
@@ -77,29 +223,158 @@ public enum Kind
             }
         }
 
+        public final ReferenceExpression reference;
         public final Kind kind;
-        public final Reference reference;
 
         public Is(String symbol, Kind kind)
         {
-            this(Reference.of(Symbol.unknownType(symbol)), kind);
+            this(Symbol.unknownType(symbol), kind);
         }
 
-        public Is(Reference reference, Kind kind)
+        public Is(ReferenceExpression reference, Kind kind)
         {
-            this.kind = kind;
             this.reference = reference;
+            this.kind = kind;
         }
 
         @Override
-        public void toCQL(StringBuilder sb, int indent)
+        public void toCQL(StringBuilder sb, CQLFormatter formatter)
         {
-            reference.toCQL(sb, indent);
+            reference.toCQL(sb, formatter);
             sb.append(" IS ").append(kind.cql);
         }
     }
 
-    class Builder
+    class And implements Conditional
```

Comment:
moved from its own class to here to be consistent with the other types in this package